src/celengine/observer.h

```diff
@@ -66,30 +69,30 @@ class ObserverFrame
     Selection getRefObject() const;
     Selection getTargetObject() const;
 
-    const ReferenceFrame* getFrame() const;
+    const ReferenceFrame::SharedConstPtr& getFrame() const;
 
     UniversalCoord convertFromUniversal(const UniversalCoord& uc, double tjd) const;
     UniversalCoord convertToUniversal(const UniversalCoord& uc, double tjd) const;
     Eigen::Quaterniond convertFromUniversal(const Eigen::Quaterniond& q, double tjd) const;
     Eigen::Quaterniond convertToUniversal(const Eigen::Quaterniond& q, double tjd) const;
 
-    static UniversalCoord convert(const ObserverFrame* fromFrame,
-                                  const ObserverFrame* toFrame,
+    static UniversalCoord convert(const ObserverFrame::SharedConstPtr &fromFrame,
```

Comment:
please be consistent with `&` positions. in some places you have Foo &f, in other Foo& f